tests/generation/test_stopping_criteria.py

```diff
@@ -107,3 +108,57 @@ def test_validate_stopping_criteria(self):
         stopping_criteria = validate_stopping_criteria(StoppingCriteriaList(), 11)
 
         self.assertEqual(len(stopping_criteria), 1)
+
+    def test_stop_string_criteria(self):
+        true_strings = [
```

Comment:
nit: I would add `stop_strings = ["<|im_end|>", "stop", "end"]` before the test cases, so we better understand the categorization of the test strings. I wondered whether the correct stop strings came from when reading this function top to bottom :D